src/ProjectOrigin.Wallet.Server/Extensions/ClaimsPrincipalExtensions.cs

```diff
@@ -0,0 +1,15 @@
+using System.Security.Claims;
+using Grpc.Core;
+
+public static class ClaimsPrincipalExtensions
+{
+    public static string GetSubject(this ServerCallContext context)
+    {
+        return context.GetHttpContext().User.GetSubject();
+    }
+
+    public static string GetSubject(this ClaimsPrincipal principal)
+    {
+        return principal.FindFirstValue(ClaimTypes.NameIdentifier) ?? throw new RpcException(new Status(StatusCode.Unauthenticated, "Subject not found on claims principal"));
```

Comment:
Is the RpcException handled somewhere? Or is this the default way when working with Grpc services?